apps/src/netsim/NetSimMessage.js

```diff
@@ -23,6 +23,29 @@ var NetSimLogger = require('./NetSimLogger');
 var logger = NetSimLogger.getSingleton();
 
 /**
+ * @typedef {Object} MessageData
+ * @property {!number} fromNodeID - sender node ID
+ * @property {!number} toNodeID - destination node ID
+ * @property {!number} simulatedBy - node ID of client simulating message
+ * @property {!string} payload - message content in a binary string
+ * @property {number} extraHopsRemaining
+ * @property {number[]} visitedNodeIDs
+ */
+
+/**
+ * @typedef {Object} MessageRow
+ * @property {number} fromNodeID - this message in-flight-from node
```

Comment:
Nit: I still figure we should mark these (fromNodeID, toNodeID, simulatedBy, base64Payload) non-nullable `!` if we're going to do so for the MessageData type above.